pkg/controllers/core/deployer_controller.go

```diff
@@ -132,8 +137,18 @@ func (r *DeployerReconciler) reconcile(ctx context.Context, log logr.Logger, dep
 	deployer.Status.ServiceName = childService.Name
 	deployer.Status.PropagateServiceStatus(&childService.Status)
 
-	deployer.Status.ObservedGeneration = deployer.Generation
+	// reconcile ingress
+	childIngress, err := r.reconcileIngress(ctx, log, deployer, childService.Name)
+	if err != nil {
+		log.Error(err, "unable to reconcile Ingress", "deployer", deployer)
+		return ctrl.Result{}, err
+	}
+	if childIngress != nil {
+		deployer.Status.PropagateIngressStatus(childIngress.Status)
+		deployer.Status.IngressName = childIngress.GetName()
```

Comment:
Setting the name before calling PropagateStatus is more common. Sometimes the Propagate method needs access to the resources name.

```suggestion
		deployer.Status.IngressName = childIngress.Name
```